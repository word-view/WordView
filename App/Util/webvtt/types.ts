export interface Cue {
    id: string;
    text: string;
    startTime: number;
    endTime: number;
}
