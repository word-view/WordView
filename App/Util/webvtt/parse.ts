import { Cue } from './types';

const { WebVTTParser } = require('webvtt-parser');

export function parseWebVTT(lyrics: string) {
    const parser = new WebVTTParser();
    const tree = parser.parse(lyrics, 'metadata');
    const cues: Cue[] = [];

    for (const cue of tree.cues) {
        cues.push(cue as Cue);
    }

    return cues;
}
