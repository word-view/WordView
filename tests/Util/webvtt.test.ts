import { parseWebVTT } from '../../App/Util/webvtt/parse'
import { kutsuNoHanabi } from '../Samples/vtt'

it(`Parse .vtt file (Kutsu no Hanabi)`, async () => {
  const parsed = parseWebVTT(kutsuNoHanabi)
  expect(parsed).toMatchSnapshot()
})
