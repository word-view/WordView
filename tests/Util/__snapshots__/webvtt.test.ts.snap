// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse .vtt file (Kutsu no Hanabi) 1`] = `
[
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 12.02,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 9.92,
    "text": "ねぇ ねぇ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "ねぇ ねぇ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 17.82,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 12.02,
    "text": "何か言おうにも言葉足らずだ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "何か言おうにも言葉足らずだ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 23.04,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 18,
    "text": "空いた口が塞がらない",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "空いた口が塞がらない",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 25.42,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 23.04,
    "text": "から",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "から",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 26.76,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 26.14,
    "text": "から",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "から",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 33.56,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 28.88,
    "text": "靴の花火",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "靴の花火",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 43.48,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 42.06,
    "text": "ねぇ ねぇ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "ねぇ ねぇ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 46.12,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 43.84,
    "text": "黙りこくっても",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "黙りこくっても",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 49.86,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 46.12,
    "text": "言葉要らずだ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "言葉要らずだ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 52.84,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 49.86,
    "text": "目って物を",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "目って物を",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 55.78,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 52.84,
    "text": "言うから",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "言うから",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 60.28,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 57,
    "text": "忘れていくことは",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "忘れていくことは",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 64.72,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 61.02,
    "text": "虫が食べ始めた結果だ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "虫が食べ始めた結果だ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 68.64,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 65.28,
    "text": "思い出の中じゃ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "思い出の中じゃ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 72.48,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 68.64,
    "text": "いつも笑ってる顔なだけ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "いつも笑ってる顔なだけ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 74.88,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 72.68,
    "text": "夕暮れた色",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "夕暮れた色",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 76.9,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 74.88,
    "text": "空を飛んで",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "空を飛んで",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 80.58,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 76.9,
    "text": "このまま大気さえ飛び出して",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "このまま大気さえ飛び出して",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 85.7,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 80.58,
    "text": "真下、次第に小さくなってくのは",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "真下、次第に小さくなってくのは",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 89.1,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 85.7,
    "text": "君の居た街だ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "君の居た街だ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 92.8,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 89.1,
    "text": "靴の先に花が咲いた",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "靴の先に花が咲いた",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 96.96,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 92.8,
    "text": "大きな火の花が咲いた",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "大きな火の花が咲いた",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 100.76,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 96.96,
    "text": "心ごと残して征こう、",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "心ごと残して征こう、",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 103.22,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 100.76,
    "text": "だなんて憶う",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "だなんて憶う",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 106.94,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 103.22,
    "text": "そんな夏が見えた",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "そんな夏が見えた",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 124.02,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 122.06,
    "text": "ねぇ ねぇ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "ねぇ ねぇ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 129.98,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 124.02,
    "text": "君を知ろうにもどっちつかずだ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "君を知ろうにもどっちつかずだ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 134.72,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 129.98,
    "text": "きっと鼻に掛けるから",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "きっと鼻に掛けるから",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 141.88,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 137.68,
    "text": "清々することなんて",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "清々することなんて",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 145.76,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 141.88,
    "text": "何にもないけど",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "何にもないけど",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 150.98,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 145.76,
    "text": "今日も空が綺麗だなぁ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "今日も空が綺麗だなぁ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 156.16,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 153.18,
    "text": "僕の食べた物",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "僕の食べた物",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 161.06,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 156.64,
    "text": "全てがきっと生への対価だ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "全てがきっと生への対価だ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 164.66,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 161.06,
    "text": "今更な僕は",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "今更な僕は",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 168.64,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 164.66,
    "text": "ヨダカにさえもなれやしない",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "ヨダカにさえもなれやしない",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 172.86,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 168.78,
    "text": "朝焼けた色 空を舞って",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "朝焼けた色 空を舞って",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 176.7,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 172.86,
    "text": "何を願うかなんて愚問だ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "何を願うかなんて愚問だ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 180.92,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 176.7,
    "text": "大人になって忘れていた",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "大人になって忘れていた",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 184.94,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 180.92,
    "text": "君を映す目が邪魔だ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "君を映す目が邪魔だ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 188.82,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 184.94,
    "text": "ずっと下で花が鳴った",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "ずっと下で花が鳴った",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 192.98,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 188.82,
    "text": "大きな火の花が鳴った",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "大きな火の花が鳴った",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 196.6,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 192.98,
    "text": "音だけでも泣いてしまう",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "音だけでも泣いてしまう",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 199.44,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 196.64,
    "text": "だなんて憶う",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "だなんて憶う",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 202.8,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 199.44,
    "text": "そんな夏を聞いた",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "そんな夏を聞いた",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 236.12,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 234.28,
    "text": "ねぇ ねぇ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "ねぇ ねぇ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 241.66,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 236.12,
    "text": "空を飛ぼうにも終わり知らずだ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "空を飛ぼうにも終わり知らずだ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 247.6,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 242.18,
    "text": "きっと君を探してしまう",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "きっと君を探してしまう",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 249.28,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 247.6,
    "text": "から",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "から",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 251.78,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 250.02,
    "text": "から",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "から",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 256.62,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 252.9,
    "text": "夕暮れた色 空を飛んで",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "夕暮れた色 空を飛んで",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 260.58,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 256.62,
    "text": "この星の今さえ抜け出して",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "この星の今さえ抜け出して",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 264.28,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 260.58,
    "text": "真下、次第に小さくて",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "真下、次第に小さくて",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 268.8,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 264.28,
    "text": "消えたのは君の居た街だ",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "消えたのは君の居た街だ",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 272.92,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 268.8,
    "text": "夏の空に花が咲いた",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "夏の空に花が咲いた",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 276.64,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 272.92,
    "text": "大きな火の花が咲いた",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "大きな火の花が咲いた",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 280.48,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 276.82,
    "text": "いつまででも泣いていたい、",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "いつまででも泣いていたい、",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 282.56,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 280.48,
    "text": "だなんて憶う",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "だなんて憶う",
        },
      ],
    },
  },
  {
    "alignment": "center",
    "direction": "horizontal",
    "endTime": 287.38,
    "id": "",
    "lineAlign": "start",
    "linePosition": "auto",
    "pauseOnExit": false,
    "positionAlign": "auto",
    "size": 100,
    "snapToLines": true,
    "startTime": 283.24,
    "text": "そんな夏が消えた",
    "textPosition": "auto",
    "tree": {
      "children": [
        {
          "type": "text",
          "value": "そんな夏が消えた",
        },
      ],
    },
  },
]
`;
